tests/unit/test_charm.py

```diff
@@ -45,9 +48,34 @@ def _get_service_configs_clone(self, relation_data):
 
 
 class LegendSdlcTestCase(legend_operator_testing.TestBaseFinosCoreServiceLegendCharm):
+
+    __metadata_yaml = None
+    __config_yaml = None
+
+    @classmethod
+    def _charm_file_content(cls, filename):
+        cls_filename = inspect.getfile(cls)
+        charm_dir = pathlib.Path(cls_filename).parents[2]
+        with open(os.path.join(charm_dir, filename), "r") as f:
+            return f.read()
+
+    @classmethod
+    def _metadata_yaml(cls):
+        if cls.__metadata_yaml is None:
+            cls.__metadata_yaml = cls._charm_file_content("metadata.yaml")
+        return cls.__metadata_yaml
+
+    @classmethod
+    def _config_yaml(cls):
+        if cls.__config_yaml is None:
+            cls.__config_yaml = cls._charm_file_content("config.yaml")
+        return cls.__config_yaml
+
     @classmethod
     def _set_up_harness(cls):
```

Comment:
Some comments explaining how and why this harness is created this way would be helpful.

Reply:
Solved.